test/greenbar/renderers/hipchat_processor_test.exs

```diff
@@ -169,4 +169,54 @@ defmodule Greenbar.Test.Test.HipChatRendererTest do
     assert expected == rendered
   end
 
+
+  test "omits a newline after attachments that are each a single line of output" do
+    directives = [%{"name" => "attachment",
+                    "fields" => [],
+                    "children" => [%{"name" => "text", "text" => "Attachment 1"}]},
+                  %{"name" => "attachment",
+                    "fields" => [],
+                    "children" => [%{"name" => "text", "text" => "Attachment 2"}]},
+                  %{"name" => "attachment",
+                    "fields" => [],
+                    "children" => [%{"name" => "text", "text" => "Attachment 3"}]},
+                  %{"name" => "attachment",
+                    "fields" => [],
+                    "children" => [%{"name" => "text", "text" => "Attachment 4"},
+                                   %{"name" => "newline"},
+                                   %{"name" => "text", "text" => "Plus a newline"}]},
+                  %{"name" => "attachment",
+                    "fields" => [],
+                    "children" => [%{"name" => "text", "text" => "Attachment 5"},
+                                   %{"name" => "newline"},
+                                   %{"name" => "text", "text" => "Plus a newline"}]},
+                  %{"name" => "attachment",
+                    "fields" => [],
+                    "children" => [%{"name" => "text", "text" => "Attachment 6"}]},
+                  %{"name" => "text",
+                    "text" => "Random text"},
+                  %{"name" => "newline"},
+                  %{"name" => "attachment",
+                    "fields" => [],
+                    "children" => [%{"name" => "text", "text" => "Attachment 7"}]}]
+
+    expected = """
+    Attachment 1<br/>
+    Attachment 2<br/>
+    Attachment 3<br/>
+    Attachment 4<br/>
+    Plus a newline<br/>
```

Comment:
I agree there are issues here, and originally went down the path of only joining multiline attachments with a newline, but there are issues in the codebase that make that difficult to do correctly as we're adding newlines to the ends of directives in a lot of places.

The main problem I was trying to fix was annoying newlines when rendering lots of single-line attachments in a list ( like `bundle list`). To fix it in the simplest way possible without affecting any other types of directives, I decided to remove the tailing newline if there was only one line of content in the attachment. While not ideal, it solves the immediate problem with minimal rendering changes to existing templates.